extensions/variants.go

```diff
@@ -140,24 +180,35 @@ type ScalarFunctionVariant struct {
 	impl        ScalarFunctionImpl
 }
 
-func (s *ScalarFunctionVariant) Name() string                     { return s.name }
-func (s *ScalarFunctionVariant) Description() string              { return s.description }
-func (s *ScalarFunctionVariant) Args() ArgumentList               { return s.impl.Args }
-func (s *ScalarFunctionVariant) Options() map[string]Option       { return s.impl.Options }
-func (s *ScalarFunctionVariant) Variadic() *VariadicBehavior      { return s.impl.Variadic }
-func (s *ScalarFunctionVariant) Deterministic() bool              { return s.impl.Deterministic }
-func (s *ScalarFunctionVariant) SessionDependent() bool           { return s.impl.SessionDependent }
-func (s *ScalarFunctionVariant) Nullability() NullabilityHandling { return s.impl.Nullability }
-func (s *ScalarFunctionVariant) URI() string                      { return s.uri }
+func (s *ScalarFunctionVariant) Name() string                { return s.name }
+func (s *ScalarFunctionVariant) Description() string         { return s.description }
+func (s *ScalarFunctionVariant) Args() ArgumentList          { return s.impl.Args }
+func (s *ScalarFunctionVariant) Options() map[string]Option  { return s.impl.Options }
+func (s *ScalarFunctionVariant) Variadic() *VariadicBehavior { return s.impl.Variadic }
+func (s *ScalarFunctionVariant) Deterministic() bool         { return s.impl.Deterministic }
+func (s *ScalarFunctionVariant) SessionDependent() bool      { return s.impl.SessionDependent }
+func (s *ScalarFunctionVariant) Nullability() NullabilityHandling {
+	if s.impl.Nullability == "" {
+		return MirrorNullability
+	}
+	return s.impl.Nullability
+}
+func (s *ScalarFunctionVariant) URI() string { return s.uri }
 func (s *ScalarFunctionVariant) ResolveType(argumentTypes []types.Type) (types.Type, error) {
-	return EvaluateTypeExpression(s.impl.Nullability, s.impl.Return, s.impl.Args, argumentTypes)
+	return EvaluateTypeExpression(s.Nullability(), s.impl.Return, s.impl.Args, argumentTypes)
 }
 func (s *ScalarFunctionVariant) CompoundName() string {
 	return s.name + ":" + s.impl.signatureKey()
 }
 func (s *ScalarFunctionVariant) ID() ID {
 	return ID{URI: s.uri, Name: s.CompoundName()}
 }
+func (s *ScalarFunctionVariant) Match(argumentTypes []types.Type) (bool, error) {
```

Comment:
In doc it would be good to give information on what kinds of errors could be return here. It's not immediately evident.

Reply:
added doc string